### PR TITLE
Add log file for UI

### PR DESCRIPTION
**Pull request by IrinaShkviro**
Duplicated log entries to Orbit.log file (`%USERPROFILE%\AppData\Roaming\OrbitProfiler\logs` on Windows and `~/.orbitprofiler/logs` on Linux))

### CODE REVIEW

File: OrbitCore/Path.cpp [hunk 1]
```diff
@@ -282,6 +282,12 @@ std::string Path::GetHome() {
   return home;
 }
 
+std::string Path::GetLogFilePath() {
+  std::string logsDir = Path::GetAppDataPath() + "logs/";
```
Comment (dimitry-) [inline OrbitCore/Path.cpp:286]: nit: use JoinPath() here and below - let it take care of '/' separators

File: OrbitBase/include/OrbitBase/Logging.h [hunk 2]
```diff
@@ -77,15 +78,28 @@
...
+    if (log_file.is_open()) {       \
+      log_file << message;          \
+      log_file.flush();             \
+    }                               \
   } while (0)
 #else
-#define PLATFORM_LOG(message) fprintf(stderr, "%s", message)
+#define PLATFORM_LOG(message) {     \
+    fprintf(stderr, "%s", message); \
+    if (log_file.is_open()) {       \
```
Comment (dimitry-) [inline OrbitBase/include/OrbitBase/Logging.h:98]: Extract this to a function and protect this call by mutex. I do not think ofstream is thread-safe

Comment (IrinaShkviro) [reply OrbitBase/include/OrbitBase/Logging.h]: Fixed it, thank you!

File: OrbitCore/Path.cpp (hunk 1 above)
Comment (IrinaShkviro) [reply OrbitCore/Path.cpp]: Added JoinPath() instead of separator

File: OrbitBase/Logging.cpp [hunk 3]
```diff
@@ -4,9 +4,18 @@
 
 #include <OrbitBase/Logging.h>
 
+absl::Mutex log_file_mutex;
 std::ofstream log_file;
 
 void InitLogFile(const std::string& path) {
   CHECK(!log_file.is_open());
```
Comment (dimitry-) [inline OrbitBase/Logging.cpp:11]: add MutexLock here as well.

File: OrbitBase/include/OrbitBase/Logging.h [hunk 4]
```diff
@@ -78,27 +79,23 @@
   } while (0 && (assertion))
 #endif
 
+extern absl::Mutex log_file_mutex;
```
Comment (dimitry-) [inline OrbitBase/include/OrbitBase/Logging.h:82]: Is this used?

File: OrbitBase/Logging.cpp [hunk 5]
```diff
@@ -4,9 +4,18 @@
 
 #include <OrbitBase/Logging.h>
 
+absl::Mutex log_file_mutex;
```
Comment (dimitry-) [inline OrbitBase/Logging.cpp:7]: I think you can make it static - all functions that use it are in this file.

File: OrbitBase/Logging.cpp (hunk 3 above)
Comment (IrinaShkviro) [reply OrbitBase/Logging.cpp:12]: Yes, thanks for noticing: I should use mutex in all places when working with the log file

File: OrbitBase/include/OrbitBase/Logging.h (hunk 4 above)
Comment (IrinaShkviro) [reply OrbitBase/include/OrbitBase/Logging.h]: Removed from header

**Pull request merged**